### ISSUE TITLE
Add a way for client to defer anchoring annotations while content is loading

### ISSUE THREAD

**Issue by robertknight** [hypothesis/client#5568]
The video player is going to switch to loading the transcript asynchronously, via an API call, after the video player frontend loads. We need to make sure the Hypothesis client doesn't try to anchor annotations until the transcript has loaded. One solution would be for the video player to avoid loading the client until the transcript has been fetched, but loading the transcript and client serially is going to be significantly slower than loading them concurrently.

**Comment by robertknight** [comment]
As part of this, we'll need to decide how the web app can communicate to Hypothesis that content _failed_ to load, and what the client should do in that case.

The most minimal solution would be that the web app can _only_ communicate loading / loaded states to the client, and errors are treated as part of the "loading" state. This would allow a subsequent loading => loaded state if the user clicks a retry button in the app's UI for example.

**Comment by robertknight** [comment]
Some possible APIs for this:

1. Hypothesis dispatches an event at `document.body` when it loads, the event has a method on it to allow the app to signal that Hypothesis should wait before anchoring. This would be similar to the `ScrollToRangeEvent.waitUntil` method that we recently added.
2. Allow the web app to add an attribute (eg. `data-hypothesis-content-loaded`) to the body element to signal that the content is not yet loaded. The web app will remove the attribute when the content is ready.
3. If we implemented [Hypothesis as a library](https://github.com/hypothesis/client/issues/5569) or some other means of getting a reference to a controller for the client, we could expose a method on the controller like `hypothesis.setContentLoaded(promise)`.
